.github/workflows/regression.yaml

```diff
@@ -147,8 +225,8 @@ jobs:
     strategy:
       matrix:
         include: [
-          {installs: "MINGW32", pkg: "mingw-w64-i686-gcc"},
-          {installs: "MINGW64", pkg: "mingw-w64-x86_64-gcc"},
+          {installs: "MINGW32", pkg: "mingw-w64-i686-gcc mingw-w64-i686-cmake mingw-w64-i686-ninja"},
+          {installs: "MINGW64", pkg: "mingw-w64-x86_64-gcc mingw-w64-x86_64-cmake mingw-w64-x86_64-ninja"},
```

Comment:
Wow, 8m30s for MINGW32 and 6m26s for MINGW64. Since merging the jobs seemed to fix the CI issue and there should be jobs to spare now maybe the CMake tests should be separate in this case? It's probably fine as is though (still faster than Travis-CI...).

Reply:
Yes. Vitaly: We should split away mingw-cmake builds like before,
otherwise it takes too much time to complete in CI

Reply:
If you really want to speed-up the CI work, let's use the `-j 3`

Reply:
Just splitting away the autotools and cmake builds should be enough.

Reply:
Does Github actually let you use multiple threads or are the extra jobs just filling disk IO time gaps? I never tried a workflow with -j# because I assumed it just wouldn't help that much, but clearly there's a decent amount of improvement.

Reply:
We shouldn't need `-j#`

Reply:
Using just the GNU Make builder, it will work in one thread only. Using the `make -j 3` is fine. GitHub CI already gives the ability to use at least two cores. When using Ninja build instead of GNU Make (that's possible to do when using CMake, Autotools is able to work with the GNU Make only, that also a pain for the BSD Make, CMake is able to work with it natively).

Reply:
At my projects where I work, using one-threaded GNU make is externally long, and so, using Ninja Build or GNU Make with -j flags, boosts the build majorly.

Reply:
And so, you can see that the -j build of every branch took less than 20 seconds while one-threaded build took more than one minute (in the case CMake build configured to build both static and shared libraries, Autotools by default builds the shared library only without building of static library)

Reply:
> GitHub CI already gives the ability to use at least two cores.

OK, took a little searching, but [here it is directly from the GitHub documentation](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners): Linux and Windows runners have two cores and Mac OS runners have 3. We absolutely *should* be using -j#, particularly for MSYS2.

Reply:
OK then. Vitaly: you should change those `-j3` to `-j2` for linux and msys, yes? Mac can stay at `-j3`

Reply:
Re: the Linux and MSYS workflows, having one or two more jobs than available cores can still help due to active jobs blocking (hence the question about IO).

Reply:
At my work colleagues suggested to use one job more than cores you have, that makes more effective task management between running jobs, so, the little example:

I built the library with -j 4 (my processor has 4 cores)
```
real	0m14,166s
user	0m31,219s
sys	0m10,187s
```

And the same library built with -j 5 on the same machine:
```
real	0m12,820s
user	0m31,105s
sys	0m9,757s
```